### ISSUE TITLE
NullReferenceException in TKCustomMapRenderer AddRoute 

### ISSUE THREAD

**Issue by CliffCawley** [TorbenK/TK.CustomMap#32]
```
  at TK.CustomMap.iOSUnified.TKCustomMapRenderer+<>c__DisplayClass2e.<AddRoute>b__2d (MapKit.MKDirectionsResponse r, Foundation.NSError e) [0x00071] in <filename unknown>:0 
  at ObjCRuntime.Trampolines+SDMKDirectionsHandler.Invoke (IntPtr block, IntPtr response, IntPtr error) [0x00019] in /Users/builder/data/lanes/2689/962a0506/source/maccore/src/build/ios/native/ObjCRuntime/Trampolines.g.cs:3594 
  at (wrapper native-to-managed) ObjCRuntime.Trampolines+SDMKDirectionsHandler:Invoke (intptr,intptr,intptr)
  at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, IntPtr principal, IntPtr delegate) [0x00005] in /Users/builder/data/lanes/2689/962a0506/source/maccore/src/UIKit/UIApplication.cs:77 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00038] in /Users/builder/data/lanes/2689/962a0506/source/maccore/src/UIKit/UIApplication.cs:61 
  at TestTkCustomMap.iOS.Application.Main (System.String[] args) [0x00008] in /Users/cliff/Documents/TestTkCustomMap/iOS/Main.cs:17 
```

Attached is a test project that reproduces it. Simply wait for it to load once, then 'double tap' the reload button.

[TestTkCustomMap.zip](https://github.com/TorbenK/TK.CustomMap/files/143593/TestTkCustomMap.zip)

**Comment by CliffCawley** [comment]
Note that I haven't tried this on Android as I'm still yet to get Android working on my end

_edit_ Scratch that, does the same thing on Android

**Comment by CliffCawley** [comment]
Reproduced this with the source so you get line numbers:

```
  at TK.CustomMap.Droid.TKCustomMapRenderer+<AddRoute>c__async2.MoveNext () [0x0011c] in /Users/cliff/Documents/TK.CustomMap/TK.CustomMap/TK.CustomMap.Android/TKCustomMapRenderer.cs:964 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/2692/e98e9627/source/mono/external/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>m__0 (System.Object state) [0x00000] in /Users/builder/data/lanes/2692/e98e9627/source/mono/external/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:994 
  at Android.App.SyncContext+<Post>c__AnonStorey0.<>m__0 () [0x00000] in /Users/builder/data/lanes/2692/e98e9627/source/monodroid/src/Mono.Android/src/Android.App/SyncContext.cs:18 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x0000b] in /Users/builder/data/lanes/2692/e98e9627/source/monodroid/src/Mono.Android/src/Java.Lang/Thread.cs:36 
  at Java.Lang.IRunnableInvoker.n_Run (IntPtr jnienv, IntPtr native__this) [0x00009] in /Users/builder/data/lanes/2692/e98e9627/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Java.Lang.IRunnable.cs:71 
  at (wrapper dynamic-method) System.Object:5c383d24-6be6-4cae-a7d0-7f8cde674656 (intptr,intptr)
```

**Comment by TorbenK** [comment]
Really appreciate the effort! Thanks!

**Comment by TorbenK** [comment]
In reference to your test project you provided me, you can now do this instead of your "hacky" workaround:

``` CSharp
mapView.RouteCalculationFinishedCommand = new Command<TKRoute>((route) =>
            {
                // Route has been calculated, populate the data
                if (this.Route.All(i => i.IsCalculated))
                {
                    this._mapView.FitToMapRegions(this.Route.Select(i => i.Bounds));
                }
            });
```

**Comment by TorbenK** [comment]
Fixed in 1.4.0
